### PR TITLE
Feature/patch pre crypto

### PR DESCRIPTION
**Pull request by cneveux**
Modification/addition of core features to start integration of HW Cryptographic driver (NXP CAAM HW module).

### CODE REVIEW

File: core/include/kernel/dt.h [hunk 1]
```diff
@@ -93,6 +95,38 @@ int dt_map_dev(const void *fdt, int offs, vaddr_t *base, size_t *size);
...
+
+/*
+ * Force secure-status = "okay" and status="disabled" for the target node.
+ *
+ * @fdt reference to the Device Tree
+ * @node is the node offset to modify
+ *
+ * Returns 0 on success or -1 on failure
+ */
+int dt_enable_secure_status(void *fdt, int node);
```
Comment (etienne-lms) [inline core/include/kernel/dt.h]: suggest to rename `dt_enable_secure_only_status()`.

File: core/kernel/dt.c [hunk 2]
```diff
@@ -51,6 +51,71 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
...
+}
+
+int dt_enable_secure_status(void *fdt, int node)
+{
+	if (dt_disable_status(fdt, node)) {
+		EMSG("Unable to disable Normal Status");
+		return -1;
+	}
+
+	if (fdt_setprop_string(fdt, node, "secure-status", "okay"))
```
Comment (etienne-lms) [inline core/kernel/dt.c]: suggest at line 112:
```
	assert(!fdt_getprop(fdt, node, "secure-status", &len) ||
	       len >= sizeof("okay"));
```

File: core/include/kernel/dt.h (hunk 1 above)
Comment (etienne-lms) [reply core/include/kernel/dt.h]: well... `dt_enable_secure_only_status()` is as confusing as `dt_enable_secure_status()`.
Discard my comment.

File: core/kernel/dt.c (hunk 2 above)
Comment (cneveux) [reply core/kernel/dt.c]: I don't understand why, could you explain?

Below is the description of the fdt_setprop_string saying that the property length will be the length of the string to set. Hence the `len >= sizeof("okay")` will never occurred.

/**
 * fdt_setprop_string - set a property to a string value
 * @fdt: pointer to the device tree blob
 * @nodeoffset: offset of the node whose property to change
 * @name: name of the property to change
 * @str: string value for the property
 *
 * fdt_setprop_string() sets the value of the named property in the
 * given node to the given string value (using the length of the
 * string to determine the new length of the property), or creates a
 * new property with that value if it does not already exist.
 *
 * This function may insert or delete data from the blob, and will
 * therefore change the offsets of some existing nodes.
 */

Comment (etienne-lms) [reply core/kernel/dt.c]: ok, you're right. discard my comment.

**Pull request merged**